### PR TITLE
Migration agent-info data to wazuhdb epic integration tests

### LINKED ISSUE #852: Integration tests for global.db access commands in Wazuh DB
# Description

The main goal of this issue is to have automated testing coverage for all the `global.db` access commands created in Wazuh DB. The list of commands is the next one:

```
    [WDB_INSERT_AGENT] = "global insert-agent %s",
    [WDB_INSERT_AGENT_GROUP] = "global insert-agent-group %s",
    [WDB_INSERT_AGENT_BELONG] = "global insert-agent-belong %s",
    [WDB_UPDATE_AGENT_NAME] = "global update-agent-name %s",
    [WDB_UPDATE_AGENT_VERSION] = "global update-agent-version %s",
    [WDB_UPDATE_AGENT_KEEPALIVE] = "global update-keepalive %s",
    [WDB_UPDATE_AGENT_STATUS] = "global update-agent-status %s",
    [WDB_UPDATE_AGENT_GROUP] = "global update-agent-group %s",
    [WDB_UPDATE_FIM_OFFSET] = "global update-fim-offset %s",
    [WDB_UPDATE_REG_OFFSET] = "global update-reg-offset %s",
    [WDB_SET_AGENT_LABELS] = "global set-labels %d %s",
    [WDB_GET_ALL_AGENTS] = "global get-all-agents start_id %d",
    [WDB_GET_AGENTS_BY_KEEPALIVE] = "global get-agents-by-keepalive condition %s %d start_id %d",
    [WDB_FIND_AGENT] = "global find-agent %s",
    [WDB_GET_AGENT_INFO] = "global get-agent-info %d",
    [WDB_GET_AGENT_LABELS] = "global get-labels %d",
    [WDB_SELECT_AGENT_NAME] = "global select-agent-name %d",
    [WDB_SELECT_AGENT_GROUP] = "global select-agent-group %d",
    [WDB_SELECT_AGENT_STATUS] = "global select-agent-status %d",
    [WDB_SELECT_KEEPALIVE] = "global select-keepalive %s %s",
    [WDB_SELECT_FIM_OFFSET] = "global select-fim-offset %d",
    [WDB_SELECT_REG_OFFSET] = "global select-reg-offset %d",
    [WDB_FIND_GROUP] = "global find-group %s",
    [WDB_SELECT_GROUPS] = "global select-groups",
    [WDB_DELETE_AGENT] = "global delete-agent %d",
    [WDB_DELETE_GROUP] = "global delete-group %s",
    [WDB_DELETE_AGENT_BELONG] = "global delete-agent-belong %d",
    [WDB_DELETE_GROUP_BELONG] = "global delete-group-belong %s"
```

At least one test case should be added for each command.

### PR DESCRIPTION
**Pull request by palaciosjeremias**
Hello team,

As part of **Migration agent-info data to wazuhdb** epic, new integration tests were implemented.
This PR add tests cases for every command received by Wazuhdb aimed to global.db
It also modifies old test cases related to agent info folder.

Closes #852

# Tests checks

- [X] Proven that tests **pass** when they have to pass
- [X] Proven that tests **fail** when they have to fail
- [x] Proven that tests have the expected behavior in **RPM and DEB**
- [x] Tested and passed in Jenkins. Build URL:
                 Ubuntu: https://ci.wazuh.info/job/test_integration/390/
                 Centos: https://ci.wazuh.info/job/test_integration/389/

Best regards.

### CODE REVIEW

**Review by DProvinciani — APPROVED** [review]
Looks good!

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 1]
```diff
@@ -42,10 +43,38 @@
 
 receiver_sockets, monitored_sockets, log_monitors = None, None, None  # Set in the fixtures
 
+def regex_match(regex, string):
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:46]:
```suggestion

def regex_match(regex, string):
```
PEP8.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 2]
```diff
@@ -42,10 +43,38 @@
...
+    regex = regex.replace(")", "")
+    string = string.replace("[", "")
+    string = string.replace("]", "")
+    string = string.replace("(", "")
+    string = string.replace(")", "")
+    return bool(re.match(regex, string))
+
+
+def insert_agents(id_offset, amount): 
+    for id in range(id_offset,id_offset+amount):
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:60]:
```suggestion
    for id in range(id_offset, id_offset+amount):
```
PEP8.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 3]
```diff
@@ -42,10 +43,38 @@
...
+        command = f'global insert-agent {{"id":{id},"name":"TestName{id}","date_add":1599223378}}'
+        receiver_sockets[0].send(command, size=True)
+        response = receiver_sockets[0].receive(size=True).decode()
+        data = response.split(" ", 1)
+        if data[0] != 'ok':
+            raise AssertionError('Unable to add agent {id}')
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:66]:
```suggestion
        assert data[0] == 'ok', f'Unable to add agent {id}'
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 4]
```diff
@@ -42,10 +43,38 @@
...
 
-# Tests
+        command = f'global update-keepalive {{"id":{id},"sync_status":"syncreq"}}'
+        receiver_sockets[0].send(command, size=True)
+        response = receiver_sockets[0].receive(size=True).decode()
+        data = response.split(" ", 1)
+        if data[0] != 'ok':
+            raise AssertionError('Unable to update agent {id}')
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:73]:
```suggestion
        assert data[0] == 'ok', f'Unable to update agent {id}'
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 5]
```diff
@@ -42,10 +43,38 @@
...
+    regex = regex.replace("*", ".*")
+    regex = regex.replace("[", "")
+    regex = regex.replace("]", "")
+    regex = regex.replace("(", "")
+    regex = regex.replace(")", "")
+    string = string.replace("[", "")
+    string = string.replace("]", "")
+    string = string.replace("(", "")
+    string = string.replace(")", "")
+    return bool(re.match(regex, string))
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:56]:
```suggestion
    return re.match(regex, string)
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 6]
```diff
@@ -61,10 +90,19 @@ def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_mod
...
+        if 'ignore' in stage and stage['ignore'] == "yes":
+            continue
+        
         expected = stage['output']
         receiver_sockets[0].send(stage['input'], size=True)
         response = receiver_sockets[0].receive(size=True).decode()
-        assert response == expected, 'Failed test case stage {}: {}'.format(test_case.index(stage) + 1, stage['stage'])
+        
+        if 'use_regex' in stage and stage['use_regex'] == 'yes':
+            match = regex_match(expected, response)
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:101]:
```suggestion
            match = True if regex_match(expected, response) else False
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 7]
```diff
@@ -61,10 +90,19 @@ def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_mod
...
+        else:
+            match = (expected == response)
+        assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}'\
+               .format(test_case.index(stage) + 1, stage['stage'], expected, response)
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:105]: If you need to know the index of `stage`, it is more efficient if you set the for like this:
```python
for index, stage in enumerate(test_case):
```

This way, you have the current index available in the variable `index`.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 8]
```diff
@@ -77,3 +115,40 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
+
+
+def test_wazuh_db_chunks(configure_sockets_environment, connect_to_sockets_module):
+    """Check that commands by chunks work properly when agents amount exceed the response maximum size"""  
+    AGENTS_CANT = 14000
+    AGENTS_OFFSET = 20
+    AGENTS_MAX = AGENTS_OFFSET+AGENTS_CANT
+    
+    #Pre insert agents
+    insert_agents(AGENTS_OFFSET, AGENTS_CANT)
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:127]: If this function is not part of the test (it seems to be a prerequisite), it should be a fixture called by this particular test.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 9]
```diff
@@ -77,3 +115,40 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
+           .format('due', status)
+   
+    #Check sync-agent-info-get chunk limit
+    command = f'global sync-agent-info-get last_id 0'
+    receiver_sockets[0].send(command, size=True)
+    response = receiver_sockets[0].receive(size=True).decode()
+    
+    status = response.split(" ",1)[0]
+    assert status == 'due', 'Failed chunks check on sync-agent-info-get. Expected: {}. Response: {}'\
+           .format('due', status)
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:154]: Refactor this to use Python inner functions.

**Review by vicferpoy — CHANGES_REQUESTED** [review]
Remove this import as it is not being used:
```python
from wazuh_testing.tools.monitoring import ManInTheMiddle
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 7 above)
Comment (palaciosjeremias) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Done.

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 8 above)
Comment (palaciosjeremias) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Done

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 9 above)
Comment (palaciosjeremias) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Done

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 10]
```diff
@@ -77,3 +116,23 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
...
+        assert status == 'due', 'Failed chunks check on < {} >. Expected: {}. Response: {}'\
+               .format(command, 'due', status)
+   
+    #Check get-all-agents chunk limit
+    test_chunk_command(f'global get-all-agents last_id 0')
+    #Check get-agents-by-keepalive chunk limit
+    test_chunk_command(f'global get-agents-by-keepalive condition > -1 last_id 0')
+    #Check sync-agent-info-get chunk limit
+    test_chunk_command(f'global sync-agent-info-get last_id 0')
+
```
Comment (vicferpoy) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:138]:
```suggestion
def test_wazuh_db_chunks(configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
    """Check that commands by chunks work properly when agents amount exceed the response maximum size""" 

    def send_chunk_command(command):
        receiver_sockets[0].send(command, size=True)
        response = receiver_sockets[0].receive(size=True).decode()
    
        status = response.split(" ", 1)[0]
        assert status == 'due', 'Failed chunks check on < {} >. Expected: {}. Response: {}'\
               .format(command, 'due', status)
   
    # Check get-all-agents chunk limit
    send_chunk_command(f'global get-all-agents last_id 0')
    # Check get-agents-by-keepalive chunk limit
    send_chunk_command(f'global get-agents-by-keepalive condition > -1 last_id 0')
    # Check sync-agent-info-get chunk limit
    send_chunk_command(f'global sync-agent-info-get last_id 0')
```

Only test functions can begin with `test_` as this is what **pytest** will parse from the file. I suggest renaming the inner function.

**Review by vicferpoy — CHANGES_REQUESTED** [review]

**Review by vicferpoy — APPROVED** [review]
LGTM!

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**